File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int curr = 1;
        int total = 0;
        for(int bed: flowerbed){
            if(bed == 0){
                curr++;
            }else{
                total += (curr - 1)/2;
                curr= 0;
            }
        }
        if (curr != 0){
            total += curr / 2;
        }
        return total >= n;
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        bed = [0] + flowerbed + [0]
        for i in range(1, len(flowerbed)+1):
            if (bed[i+1] + bed[i-1] + bed[i]) == 0:
                n -= 1
                bed[i] = 1
        if n <= 0: return True
        return False